### PR TITLE
Calculate DaysMonth, WxMonth, WxYear

### PR DESCRIPTION
**Pull request by rajadain**
Rather than packing this heavy data in JSON that gets stored everywhere, we just calculate it during initialization.
## Testing Instructions

Execute this snippet https://gist.github.com/kdeloach/e37a0d3dd805b86d4a427e602df16423 with this file: [mapshed_data.json](https://github.com/kdeloach/gwlf-e/files/278022/mapshed_data.json.txt)

Connects https://github.com/WikiWatershed/model-my-watershed/issues/1319

### CODE REVIEW

**Comment by rajadain** [comment]
Updated to only calculate data if it is missing. Added testing instructions.

File: gwlfe/datamodel.py [hunk 1]
```diff
@@ -105,6 +107,31 @@ def defaults(self):
...
+        model = self.__dict__
+        if 'WxYrBeg' in model and 'WxYrEnd' in model:
+            begyear = model['WxYrBeg']
+            endyear = model['WxYrEnd']
+            year_range = endyear - begyear + 1
+            month_abbr = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
+                          'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
+
+            if 'DaysMonth' not in model:
+                model['DaysMonth'] = np.ones((year_range, 12),
```
Comment (kdeloach) [inline gwlfe/datamodel.py:120]: I don't think it's terribly important to initialize each month to have 31 days when it gets assigned the correct number in the next loop. `np.zeros` may be good enough.

File: gwlfe/datamodel.py [hunk 2]
```diff
@@ -105,6 +107,31 @@ def defaults(self):
...
+                                             dtype=np.int) * 31
+                for y in range(year_range):
+                    year = begyear + y
+                    for m in range(12):
+                        model['DaysMonth'][y][m] = monthrange(year, m + 1)[1]
```
Comment (kdeloach) [inline gwlfe/datamodel.py:125]: Great solution.

**Comment by kdeloach** [comment]
+1 Good compromise.

**Pull request merged**